stl/inc/yvals_core.h

```diff
@@ -799,7 +799,21 @@
 #define _DEPRECATE_STDEXT_HASH_UPPER_BOUND
 #endif // ^^^ warning disabled ^^^
 
-// next warning number: STL4024
+// P0966R1 [depr.string.capacity]
+#if _HAS_CXX20 && !defined(_SILENCE_CXX20_STRING_RESERVE_DEFAULT_ARGUMENT_DEPRECATION_WARNING) \
+    && !defined(_SILENCE_ALL_CXX20_DEPRECATION_WARNINGS)
+#define _CXX20_DEPRECATE_STRING_RESERVE_DEFAULT_ARGUMENT                                                             \
+    [[deprecated("warning STL4024: "                                                                                 \
+                 "std::string::reserve() with the default arument 0 is deprecated in C++20. "                        \
+                 "If the intention is to shrink the string's capacity, use std::string::shrink_to_fit(). Otherwise, "\
+                 "provide an argument to std::string::reserve(). "                                                   \
+                 "You can define _SILENCE_CXX20_STRING_RESERVE_DEFAULT_ARGUMENT_DEPRECATION_WARNING"                 \

```

Comment:
There is no space before the `"or"` on the next line.

Reply:
I also suspect that this needs to be clang-formatted since I expect it to put a space between the quote and backslash. We'll see!